operator/src/lib.rs

```diff
@@ -453,11 +451,14 @@ fn build_pod(
             containers: vec![Container {
                 image: Some(format!(
                     "stackable/kafka:{}",
-                    serde_json::json!(resource.spec.version).as_str().unwrap()
+                    &resource.spec.version.get_fully_qualified_version()
```

Comment:
No need to dereference the `String`.
```suggestion
                    resource.spec.version.get_fully_qualified_version()
```

Reply:
fixed